core/src/sigverifier/bls_sigverifier.rs

```diff
@@ -44,6 +44,12 @@ impl SigVerifier for BLSSigVerifier {
         for packet in packet_batches.iter().flatten() {
             stats_updater.received += 1;
 
+            if packet.meta().discard() {
+                // This can happen if a cert is forwarded by multiple sources.
```

Comment:
is there any other reason we would mark this packet for discard?

Reply:
We could discard for any other verification error as well. In the future we could also mark discard for BLS verification failures, when that happens we would adjust the stats accordingly. @samkim-crypto